### ISSUE TITLE
Create a code generation demonstrator

### ISSUE THREAD

**Issue by bergmanngabor** [IncQueryLabs/MD_plugin_skeleton#9]
Generate Python code from components and their connections

**Comment by bergmanngabor** [comment]
First demoable version available on branch `issues/9`

**Comment by istvanrath** [comment]
[com.incquerylabs.magicdraw.plugin.example-0.1.0.zip](https://github.com/IncQueryLabs/MD_plugin_skeleton/files/3315367/com.incquerylabs.magicdraw.plugin.example-0.1.0.zip)

**Comment by bergmanngabor** [comment]
Dependency - this V4MD build: [com.incquerylabs.v4md-2.1.3-plugin.zip](https://github.com/IncQueryLabs/MD_plugin_skeleton/files/3315073/com.incquerylabs.v4md-2.1.3-plugin.zip)

Sample project to experiment on: [CodegenExample.zip](https://github.com/IncQueryLabs/MD_plugin_skeleton/files/3315080/CodegenExample.zip)

Jupyter notebook to play around with the code generated from the above project: [pygen_test.ipynb.zip](https://github.com/IncQueryLabs/MD_plugin_skeleton/files/3315085/pygen_test.ipynb.zip)

**Comment by JMZ-Saf** [comment]
Thanks.

Any installation procedure?

**Comment by istvanrath** [comment]
0. Use Cameo Systems Modeler 19.0 SP1 or SP2 on Windows
1. Install both plugins (com.incquerylabs.magicdraw.plugni.example and v4md) via Resource Manager (Help | Resource/Plugin Manager | Import | select _both_ zip files)
2. Restart Cameo Systems Modeler
3. After restart, open sample project (unzip CodegenExample.zip and open the .mdzip inside)
4. From Example menu, select Generate Python Code
5. Select a source package, we recommend "Example Solution"
6. Select a target directory
7. Once the generation is finished, you should see python files. If you copy the .ipynb from above, and open that locally as a Jupyter notebook, you should be able to test the generated code.

**Comment by lunkpeter** [comment]
EDIT - I have updated the plugin zip file, it works on Mac as well

**Comment by istvanrath** [comment]
In the Jupyter notebook, you should see something similar:

![Screenshot 2019-06-23 at 21 18 50](https://user-images.githubusercontent.com/1567295/59981572-9b899a00-9605-11e9-8e20-65381b7c7170.png)

**Comment by JMZ-Saf** [comment]
I've dowload the Desktop version and install it, as well as CAMEO SYstems Modeler integration and FlexxNet licence import. I'v tried to launch the Example tranformation and it works.
The I've started to manage existing querry for the python transformation by my own and I'cant reached ou the Built MagicDraw Text editor (as explain into the Desktop Quick Start Guide). So then I've tried to get the External VQL Editor (the Eclipse VIATRA Editor)....but still not reaching the capability to update the querries.

**Comment by istvanrath** [comment]
Hi, this plug-in does not support editing the queries via IncQuery Desktop. We need to work on that. Currently, all the queries are "baked into" the code.

**Comment by JMZ-Saf** [comment]
Ok got it. So that How can I create a new plugin?

**Comment by istvanrath** [comment]
https://docs.nomagic.com/display/MD190/Developing+plugins+using+IDE

**Comment by bergmanngabor** [comment]
Merged into master by commit 558324aa2e861fdb861939f15c202ff3e89903db
